apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/environments/TargetFilterUniquenessIndicator.tsx

```diff
@@ -0,0 +1,186 @@
+import type { Target } from "@ctrlplane/db/schema";
+import type { UseQueryResult } from "@tanstack/react-query";
+import type { Node } from "reactflow";
+import React, { useMemo } from "react";
+import Link from "next/link";
+import { IconAlertTriangle, IconCheck, IconLoader } from "@tabler/icons-react";
+import { useQueries } from "@tanstack/react-query";
+import LZString from "lz-string";
+
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@ctrlplane/ui/tooltip";
+
+import { api } from "~/trpc/react";
+
+type UniqueFilterResult = {
+  isUnique: boolean;
+  overlaps: {
+    nodeId: string;
+    overlappingTargets: Target[];
+  }[];
+};
+
+export const useTargetFilterUniqueness = (
+  nodes: Node[],
+  workspaceId: string,
+  currentNode: Node,
+): UniqueFilterResult | null => {
+  const utils = api.useUtils();
+  const otherNodes = useMemo(
+    () => nodes.filter((node) => node.id !== currentNode.id),
+    [nodes, currentNode.id],
+  );
+
+  const overlappingQueries: UseQueryResult<Target[], Error>[] = useQueries({
+    queries: otherNodes.map((node) => ({
+      queryKey: [
+        "target",
+        "overlappingTargets",
+        workspaceId,
+        JSON.stringify(currentNode.data.targetFilter),
+        JSON.stringify(node.data.targetFilter),
+        node.id,
+      ],
+      queryFn: () =>
+        utils.target.byWorkspaceId.overlappingTargets.fetch({
+          workspaceId,
+          filterA: currentNode.data.targetFilter,
+          filterB: node.data.targetFilter,
+        }),
+      enabled: Boolean(
+        workspaceId && currentNode.data.targetFilter && node.data.targetFilter,
+      ),
+      staleTime: 5 * 60 * 1000,
+      cacheTime: 30 * 60 * 1000,
```

Comment:
is `ms` lib installed? what are the defaults?

Reply:
Sent in discord.

Reply:
ms lib?